src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -70,6 +71,9 @@ public void DeviceSetup ()
 				SetAdbLogcatBufferSize (64);
 				CreateGuestUser (GuestUserName);
 			}
+			if (string.IsNullOrEmpty (DeviceAbi)) {
```

Comment:
I'm not sure about this.  These tests are failing due to an issue with the auto-retry mechanism (that I am struggling to find a solution for).  I think this just hides that failure instead of fixing it.

https://github.com/xamarin/xamarin-android/pull/8021
https://github.com/xamarin/xamarin-android/pull/8107

Reply:
Yeah this change only better handles a potential NRE, the root cause seems to be that we somehow lose connection to the device. I think that can be investigated in any future stability improvement attempts.

Reply:
I am against adding this, I think it will hide real failures that I am still trying to find the root cause of.

Reply:
I'm not sure how throwing null / invalid format here is better than failing during installation or later in the tests. I'll remove it though, perhaps if we end up with a null/empty ABI in this block we should try to refresh the device info and throw here explicitly, rather than waiting for some other function to use the null/empty value.

Reply:
I agree that throwing `null` here isn't better, but hiding this error *right now* will make it harder, if not impossible, for me to continue investigating the root cause of the error.

It does not seem to be simply losing device connectivity, it seems to be an issue where the logic in `[OneTimeSetup]` is not run on the Auto-Retry, which is a larger issue than just the `DeviceAbi` issue.

Reply:
Ahh interesting ok makes sense, I didn't realize there was suspicious behavior around NUnit 3 fixture setup with the auto retry. The behavior with that has always been a bit odd, it seems that it runs before any logging mechanisms are hooked up as well so it's near impossible to debug. Might be worth playing around with removing it.

I backed those changes out, I think this PR should be good to go now.